test/2-config-test.js

```diff
@@ -418,6 +418,25 @@ vows.describe('Test suite for node-config')
       assert.equal(moduleConfig.parm2, 2000);
     },
 
+    'Preserves existing setModuleDefaults after Config.get() behavior': function() {
```

Comment:
This is not a clear test name. Instead of referring to preserving existing behavior, say what the behavior is.  Explain what the purpose or behavior of calling setModuleDefaults after `config.get()` is.

If I understand, it's the behavior the config should be remain mutable via setModuleDefaults if `ALLOW_CONFIG_MUTATIONS` is true. 

Otherwise I trust you that this is a regression and agree that things should be mutable when that env var is set, so self-merge after the test name is updated.

Reply:
The problem is that I “simplified” a merge into a swap, which is only visible to the user if mutability is on. I never use mutability so I just saw overwrought object merges and removed all of them (turns out I can still remove most, except this one). 

I would like to propose mutable for the chopping block but only after we make deferred and setModuleDefaults work more cleanly.